### PR TITLE
Add unit tests for SizeEnforcingInputStream

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Running "./gradlew build :grpc-all:jacocoTestReport"  seems to indicate that the methods in this class were not well covered.  I am adding these really simple unit tests to cover it.

### CODE REVIEW

File: core/src/test/java/io/grpc/internal/MessageDeframerTest.java [hunk 1]
```diff
@@ -211,6 +219,121 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
     verifyNoMoreInteractions(listener);
   }
 
+  @Test
+  public void sizeEnforcingInputStream_readByteBelowLimit() throws IOException {
+    ByteArrayInputStream in = new ByteArrayInputStream("foo".getBytes(Charsets.UTF_8));
```
Comment (nmittler) [inline core/src/test/java/io/grpc/internal/MessageDeframerTest.java]: You're always using the same input stream ... maybe create it in a `@Setup` method? This should get rid of the warning as well.

Comment (nmittler) [inline core/src/test/java/io/grpc/internal/MessageDeframerTest.java]: Actually, it won't get rid of the warning :/

Comment (nmittler) [inline core/src/test/java/io/grpc/internal/MessageDeframerTest.java]: Maybe just add a call to `stream.close()` at the end of each method?

Comment (carl-mastrangelo) [inline core/src/test/java/io/grpc/internal/MessageDeframerTest.java]: Done.

**Comment by nmittler** [comment]
@carl-mastrangelo LGTM

**Pull request merged**